`http.server.duration` metric description is wrong

The metric description for `http.server.duration` should be `Measures the duration of inbound HTTP requests` while we're using `measures the duration of the inbound HTTP request`.

This could cause an error like this (from my environment)

```
2024-02-21 14:00:05 2024-02-21T06:00:05.268Z    error   prometheusexporter@v0.93.0/log.go:23    error gathering metrics: collected metric http_server_duration_milliseconds label:{name:"deployment_environment" value:"local"} label:{name:"http_flavor" value:"1.0"} label:{name:"http_host" value:"[127.0.0.1:10000](http://127.0.0.1:10000/)"} label:{name:"http_method" value:"POST"} label:{name:"http_scheme" value:"http"} label:{name:"http_server_name" value:"[0.0.0.0](http://0.0.0.0/)"} label:{name:"http_status_code" value:"200"} label:{name:"job" value:"featbit-das"} label:{name:"label1" value:"featbit"} label:{name:"net_host_port" value:"10000"} histogram:{sample_count:3 sample_sum:44 bucket:{cumulative_count:0 upper_bound:0} bucket:{cumulative_count:0 upper_bound:5} bucket:{cumulative_count:0 upper_bound:10} bucket:{cumulative_count:3 upper_bound:25} bucket:{cumulative_count:3 upper_bound:50} bucket:{cumulative_count:3 upper_bound:75} bucket:{cumulative_count:3 upper_bound:100} bucket:{cumulative_count:3 upper_bound:250} bucket:{cumulative_count:3 upper_bound:500} bucket:{cumulative_count:3 upper_bound:750} bucket:{cumulative_count:3 upper_bound:1000} bucket:{cumulative_count:3 upper_bound:2500} bucket:{cumulative_count:3 upper_bound:5000} bucket:{cumulative_count:3 upper_bound:7500} bucket:{cumulative_count:3 upper_bound:10000}} has help "measures the duration of the inbound HTTP request" but should have "Measures the duration of inbound HTTP requests."
2024-02-21 14:00:05     {"kind": "exporter", "data_type": "metrics", "name": "prometheus"}
2024-02-21 14:00:05 [github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter.(*promLogger).Println](http://github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter.(*promLogger).Println)
2024-02-21 14:00:05     [github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter@v0.93.0/log.go:23](http://github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter@v0.93.0/log.go:23)
2024-02-21 14:00:05 [github.com/prometheus/client_golang/prometheus/promhttp.HandlerForTransactional.func1](http://github.com/prometheus/client_golang/prometheus/promhttp.HandlerForTransactional.func1)
2024-02-21 14:00:05     [github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/http.go:144](http://github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/http.go:144)
2024-02-21 14:00:05 net/http.HandlerFunc.ServeHTTP
2024-02-21 14:00:05     net/http/server.go:2136
2024-02-21 14:00:05 net/http.(*ServeMux).ServeHTTP
2024-02-21 14:00:05     net/http/server.go:2514
2024-02-21 14:00:05 [go.opentelemetry.io/collector/config/confighttp.(*decompressor).ServeHTTP](http://go.opentelemetry.io/collector/config/confighttp.(*decompressor).ServeHTTP)
2024-02-21 14:00:05     [go.opentelemetry.io/collector/config/confighttp@v0.93.0/compression.go:147](http://go.opentelemetry.io/collector/config/confighttp@v0.93.0/compression.go:147)
2024-02-21 14:00:05 [go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.(*middleware).serveHTTP](http://go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.(*middleware).serveHTTP)
2024-02-21 14:00:05     [go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.47.0/handler.go:225](http://go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.47.0/handler.go:225)
2024-02-21 14:00:05 [go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.NewMiddleware.func1.1](http://go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.NewMiddleware.func1.1)
2024-02-21 14:00:05     [go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.47.0/handler.go:83](http://go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.47.0/handler.go:83)
2024-02-21 14:00:05 net/http.HandlerFunc.ServeHTTP
2024-02-21 14:00:05     net/http/server.go:2136
2024-02-21 14:00:05 [go.opentelemetry.io/collector/config/confighttp.(*clientInfoHandler).ServeHTTP](http://go.opentelemetry.io/collector/config/confighttp.(*clientInfoHandler).ServeHTTP)
2024-02-21 14:00:05     [go.opentelemetry.io/collector/config/confighttp@v0.93.0/clientinfohandler.go:28](http://go.opentelemetry.io/collector/config/confighttp@v0.93.0/clientinfohandler.go:28)
2024-02-21 14:00:05 net/http.serverHandler.ServeHTTP
2024-02-21 14:00:05     net/http/server.go:2938
2024-02-21 14:00:05 net/http.(*conn).serve
2024-02-21 14:00:05     net/http/server.go:2009
```

- https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-python-contrib+HTTP_SERVER_DURATION&type=code
- https://github.com/open-telemetry/opentelemetry-demo/pull/738

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,13 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
-### Fixed
 - `opentelemetry-instrumentation-celery` Allow Celery instrumentation to be installed multiple times
   ([#2342](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2342))
 - Align gRPC span status codes to OTEL specification
   ([#1756](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1756))
 - `opentelemetry-instrumentation-flask` Add importlib metadata default for deprecation warning flask version
   ([#2297](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2297))
+- Ensure all http.server.duration metrics have the same description
+  ([#2151](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2298))
 
 ## Version 1.23.0/0.44b0 (2024-02-23)
 

--- a/instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py
@@ -207,7 +207,7 @@ async def middleware(request, handler):
     duration_histogram = meter.create_histogram(
         name=MetricInstruments.HTTP_SERVER_DURATION,
         unit="ms",
-        description="measures the duration of the inbound HTTP request",
+        description="Measures the duration of inbound HTTP requests",
     )
 
     active_requests_counter = meter.create_up_down_counter(

--- a/instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py
@@ -207,7 +207,7 @@ async def middleware(request, handler):
     duration_histogram = meter.create_histogram(
         name=MetricInstruments.HTTP_SERVER_DURATION,
         unit="ms",
-        description="Measures the duration of inbound HTTP requests",
+        description="Duration of HTTP client requests.",
     )
 
     active_requests_counter = meter.create_up_down_counter(

--- a/instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py
@@ -492,7 +492,7 @@ class OpenTelemetryMiddleware:
         self.duration_histogram = self.meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="measures the duration of the inbound HTTP request",
+            description="Measures the duration of inbound HTTP requests",
         )
         self.server_response_size_histogram = self.meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_RESPONSE_SIZE,

--- a/instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py
@@ -492,7 +492,7 @@ class OpenTelemetryMiddleware:
         self.duration_histogram = self.meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="Measures the duration of inbound HTTP requests",
+            description="Duration of HTTP client requests.",
         )
         self.server_response_size_histogram = self.meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_RESPONSE_SIZE,

--- a/instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py
@@ -322,7 +322,7 @@ class DjangoInstrumentor(BaseInstrumentor):
         _DjangoMiddleware._duration_histogram = meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="measures the duration of the inbound http request",
+            description="Measures the duration of inbound HTTP requests",
         )
         _DjangoMiddleware._active_request_counter = meter.create_up_down_counter(
             name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py
@@ -322,7 +322,7 @@ class DjangoInstrumentor(BaseInstrumentor):
         _DjangoMiddleware._duration_histogram = meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="Measures the duration of inbound HTTP requests",
+            description="Duration of HTTP client requests.",
         )
         _DjangoMiddleware._active_request_counter = meter.create_up_down_counter(
             name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py
@@ -268,7 +268,7 @@ class _InstrumentedFalconAPI(getattr(falcon, _instrument_app)):
         self.duration_histogram = self._otel_meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="Measures the duration of inbound HTTP requests",
+            description="Duration of HTTP client requests.",
         )
         self.active_requests_counter = self._otel_meter.create_up_down_counter(
             name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py
@@ -268,7 +268,7 @@ class _InstrumentedFalconAPI(getattr(falcon, _instrument_app)):
         self.duration_histogram = self._otel_meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="measures the duration of the inbound HTTP request",
+            description="Measures the duration of inbound HTTP requests",
         )
         self.active_requests_counter = self._otel_meter.create_up_down_counter(
             name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py
@@ -506,7 +506,7 @@ class _InstrumentedFlask(flask.Flask):
         duration_histogram = meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="measures the duration of the inbound HTTP request",
+            description="Measures the duration of inbound HTTP requests",
         )
         active_requests_counter = meter.create_up_down_counter(
             name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,
@@ -612,7 +612,7 @@ class FlaskInstrumentor(BaseInstrumentor):
             duration_histogram = meter.create_histogram(
                 name=MetricInstruments.HTTP_SERVER_DURATION,
                 unit="ms",
-                description="measures the duration of the inbound HTTP request",
+                description="Measures the duration of inbound HTTP requests",
             )
             active_requests_counter = meter.create_up_down_counter(
                 name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py
@@ -506,7 +506,7 @@ class _InstrumentedFlask(flask.Flask):
         duration_histogram = meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="Measures the duration of inbound HTTP requests",
+            description="Duration of HTTP client requests.",
         )
         active_requests_counter = meter.create_up_down_counter(
             name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py
@@ -612,7 +612,7 @@ class FlaskInstrumentor(BaseInstrumentor):
             duration_histogram = meter.create_histogram(
                 name=MetricInstruments.HTTP_SERVER_DURATION,
                 unit="ms",
-                description="Measures the duration of inbound HTTP requests",
+                description="Duration of HTTP client requests.",
             )
             active_requests_counter = meter.create_up_down_counter(
                 name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-pyramid/src/opentelemetry/instrumentation/pyramid/callbacks.py
+++ b/instrumentation/opentelemetry-instrumentation-pyramid/src/opentelemetry/instrumentation/pyramid/callbacks.py
@@ -140,7 +140,7 @@ def trace_tween_factory(handler, registry):
     duration_histogram = meter.create_histogram(
         name=MetricInstruments.HTTP_SERVER_DURATION,
         unit="ms",
-        description="Measures the duration of inbound HTTP requests",
+        description="Duration of HTTP client requests.",
     )
     active_requests_counter = meter.create_up_down_counter(
         name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-pyramid/src/opentelemetry/instrumentation/pyramid/callbacks.py
+++ b/instrumentation/opentelemetry-instrumentation-pyramid/src/opentelemetry/instrumentation/pyramid/callbacks.py
@@ -140,7 +140,7 @@ def trace_tween_factory(handler, registry):
     duration_histogram = meter.create_histogram(
         name=MetricInstruments.HTTP_SERVER_DURATION,
         unit="ms",
-        description="measures the duration of the inbound HTTP request",
+        description="Measures the duration of inbound HTTP requests",
     )
     active_requests_counter = meter.create_up_down_counter(
         name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-tornado/src/opentelemetry/instrumentation/tornado/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-tornado/src/opentelemetry/instrumentation/tornado/__init__.py
@@ -296,7 +296,7 @@ def _create_server_histograms(meter) -> Dict[str, Histogram]:
         MetricInstruments.HTTP_SERVER_DURATION: meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="measures the duration outbound HTTP requests",
+            description="Measures the duration of inbound HTTP requests",
         ),
         MetricInstruments.HTTP_SERVER_REQUEST_SIZE: meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_REQUEST_SIZE,

--- a/instrumentation/opentelemetry-instrumentation-tornado/src/opentelemetry/instrumentation/tornado/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-tornado/src/opentelemetry/instrumentation/tornado/__init__.py
@@ -296,7 +296,7 @@ def _create_server_histograms(meter) -> Dict[str, Histogram]:
         MetricInstruments.HTTP_SERVER_DURATION: meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="Measures the duration of inbound HTTP requests",
+            description="Duration of HTTP client requests.",
         ),
         MetricInstruments.HTTP_SERVER_REQUEST_SIZE: meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_REQUEST_SIZE,

--- a/instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py
@@ -507,7 +507,7 @@ class OpenTelemetryMiddleware:
         self.duration_histogram = self.meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="Measures the duration of inbound HTTP requests",
+            description="Duration of HTTP client requests.",
         )
         self.active_requests_counter = self.meter.create_up_down_counter(
             name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

--- a/instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py
+++ b/instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py
@@ -507,7 +507,7 @@ class OpenTelemetryMiddleware:
         self.duration_histogram = self.meter.create_histogram(
             name=MetricInstruments.HTTP_SERVER_DURATION,
             unit="ms",
-            description="measures the duration of the inbound HTTP request",
+            description="Measures the duration of inbound HTTP requests",
         )
         self.active_requests_counter = self.meter.create_up_down_counter(
             name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,

